static/app/views/settings/organizationIntegrations/integrationReposAddRepository.tsx

```diff
@@ -126,22 +126,30 @@ export function IntegrationReposAddRepository({
 
   return (
     <DropdownWrapper>
-      <DropdownAutoComplete
-        items={dropdownItems}
-        onSelect={addRepo}
-        onChange={e => setSearch(e.target.value)}
-        emptyMessage={t('Please enter a repository name')}
-        noResultsMessage={t('No repositories found')}
+      <CompactSelect
+        size="xs"
+        menuWidth={250}
+        options={dropdownItems}
+        onChange={addRepo}
+        disabled={false}
+        menuTitle={t('Repositories')}
+        triggerLabel={t('Add Repository')}
+        emptyMessage={
+          query.isFetching
+            ? t('Searching\u2026')
+            : debouncedSearch
+              ? t('No repositories found')
+              : t('Please enter a repository name')
+        }
         searchPlaceholder={t('Search Repositories')}
-        busy={query.isFetching}
-        alignMenu="right"
-      >
-        {({isOpen}) => (
-          <DropdownButton isOpen={isOpen} size="xs" busy={adding}>
-            {t('Add Repository')}
-          </DropdownButton>
-        )}
-      </DropdownAutoComplete>
+        loading={query.isFetching}
```

Comment:
note: since the loading spinner renders in the `MenuHeader`, we _have to_ specify a `menuTitle` or it just won’t show up at all. overall just very gnarly DX.